### PR TITLE
mne.pick_info cleans info['bads'] when bad_channels are excluded

### PR DESCRIPTION
**Pull request by jmontoyam**
mne.pick_info cleans info['bads'] when bad_channels are excluded

### CODE REVIEW

File: mne/io/pick.py [hunk 1]
```diff
@@ -296,6 +296,12 @@ def pick_info(info, sel=[], copy=True):
...
         raise ValueError('No channels match the selection.')
 
+    # Check if bads_channels are included, otherwise
+    # remove info['bads']
+    idx_bad_channels = pick_channels(info['ch_names'], info['bads'])
+    if not np.in1d(idx_bad_channels, sel).all():
+        info['bads'] = []
+
     info['chs'] = [info['chs'][k] for k in sel]
     info['ch_names'] = [info['ch_names'][k] for k in sel]
```
Comment (agramfort) [inline mne/io/pick.py]: I would insert here:

```
info['bads'] = [ch for ch in info['bads'] if ch in info['ch_names']]
```

**Comment by agramfort** [comment]
ping @jmontoyam

File: mne/io/pick.py (hunk 1 above)
Comment (dengemann) [inline mne/io/pick.py]: yes +1 this our local fix.

Comment (dengemann) [inline mne/io/pick.py]: Basically it can replace the line that's here. If there is not bad channel at all it will return an empty list.

**Comment by jmontoyam** [comment]
Thank you for your comments @agramfort, @dengemann .
Ready for review.

**Comment by dengemann** [comment]
@LGTM! I'm happy when Travis is happy too.

File: mne/io/tests/test_pick.py [hunk 2]
```diff
@@ -126,3 +128,22 @@ def test_picks_by_channels():
...
+    # select eeg channels excluding bads
+    picks_eeg = pick_types(raw.info, meg=False, eeg=True, exclude='bads')
+    info_eeg = pick_info(raw.info, picks_eeg)
+
+    # select meg channels excluding bads
+    picks_meg = pick_types(raw.info, meg='mag', eeg=False, exclude='bads')
+    info_meg = pick_info(raw.info, picks_meg)
+
+    assert_equal(len(info_eeg['bads']), 0)
+    assert_equal(len(info_meg['bads']), 0)
```
Comment (agramfort) [inline mne/io/tests/test_pick.py]: Are you sure there are bad channels in the first place?

You should not need testing for this test. Use test_raw.fif in data folder

**Comment by jmontoyam** [comment]
Hi @agramfort.

I have verified that the raw_fname that I have used for the test contains bads channels (eeg and meg bads channels). I ran the test locally and it worked ok.

Sorry, but I don't understand what you mean when you say

1) You should not need testing for this test. 
Is this test not necessary at all?...I understood that every change that I made to the mne python code should be tested.

2) Use test_raw.fif in data folder
The file that you mention (mne/io/test/data/test_raw.fif) doesn't have "bads channels",
that's the reason why I'm using for the test the following data:
mne.datasets import testing

Should I used another data for testing?

Thank you very much for your comments and suggestions!

**Comment by Eric89GXL** [comment]
But you can load the `mne/io/test/data/test_raw.fif` file, add some channels as bad, and then proceed with your tests, right? I think that's the idea.

**Comment by agramfort** [comment]
Yes

**Comment by jmontoyam** [comment]
Thanks for the suggestion @agramfort, @Eric89GXL, I will follow your advice.
Nevertheless, just out of curiosity and for learning, why is not a good idea
to use data from mne.datasets.testing?...I thought this data was ideal for testing.

**Comment by Eric89GXL** [comment]
It is good for testing, but if you can get away with using the files shipped with the repo it's slightly better. I think there are some systems where the testing dataset is not downloaded (e.g., during Debian builds) so we want to maximize the coverage we can get without the dataset. It's also not tiny, so some users might prefer not to download it at all.

**Comment by jmontoyam** [comment]
@agramfort, @Eric89GXL,
The test has been updated following your advices.
Ready for review.

File: mne/io/tests/test_pick.py [hunk 3]
```diff
@@ -11,6 +11,10 @@
 from mne.io.constants import FIFF
 from mne.datasets import testing
 from mne.forward.tests import test_forward
+from mne.io import Raw
+from mne import pick_info
+import mne
```
Comment (Eric89GXL) [inline mne/io/tests/test_pick.py]: We try to combine imports if possible, so whatever you use from `mne` namespace here can be imported on line `6` (or actually you'll need to expand that list onto the next line) and also line 15 can be combined with that list as well

File: mne/io/tests/test_pick.py [hunk 4]
```diff
@@ -126,3 +130,38 @@ def test_picks_by_channels():
     pick_list2 = _picks_by_type(raw.info, meg_combined=True)
     assert_equal(len(pick_list), len(pick_list2))
     assert_equal(pick_list2[0][0], 'mag')
+
+
+def test_clean_info_bads():
```
Comment (Eric89GXL) [inline mne/io/tests/test_pick.py]: Please add one-liner description of the test

**Comment by Eric89GXL** [comment]
other than my minor gripes LGTM

**Comment by agramfort** [comment]
merged by rebase after a tiny cosmit on the test.

thanks @jmontoyam !

**Pull request merged**